plutus-ledger-api/src/PlutusLedgerApi/V1/Interval.hs

```diff
@@ -267,46 +326,67 @@ overlaps l r = not $ isEmpty (l `intersection` r)
 {-# INLINABLE intersection #-}
 -- | 'intersection a b' is the largest interval that is contained in 'a' and in
 --   'b', if it exists.
-intersection :: Ord a => Interval a -> Interval a -> Interval a
+intersection :: (Enum a, Ord a) => Interval a -> Interval a -> Interval a
 intersection (Interval l1 h1) (Interval l2 h2) = Interval (max l1 l2) (min h1 h2)
 
 {-# INLINABLE hull #-}
 -- | 'hull a b' is the smallest interval containing 'a' and 'b'.
-hull :: Ord a => Interval a -> Interval a -> Interval a
+hull :: (Enum a, Ord a) => Interval a -> Interval a -> Interval a
 hull (Interval l1 h1) (Interval l2 h2) = Interval (min l1 l2) (max h1 h2)
 
 {-# INLINABLE contains #-}
 {- | @a `contains` b@ is true if the 'Interval' @b@ is entirely contained in
 @a@. That is, @a `contains` b@ if for every entry @s@, if @member s b@ then
 @member s a@.
 -}
-contains :: Ord a => Interval a -> Interval a -> Bool
+contains :: (Enum a, Ord a) => Interval a -> Interval a -> Bool
+-- Everything contains the empty interval
+contains _ i2 | isEmpty i2 = True
+-- Nothing is contained in the empty interval (except the empty interval,
+-- which is already handled)
+contains i1 _ | isEmpty i1 = False
+-- Otherwise we check the endpoints. This doesn't work for empty intervals,
+-- hence the cases above.
 contains (Interval l1 h1) (Interval l2 h2) = l1 <= l2 && h2 <= h1
 
 {-# INLINABLE isEmpty #-}
-{- | Check if an 'Interval' is empty.
-
-There can be many empty intervals, given a set of values.
-Two 'Interval's being empty does not imply that they are `Haskell.Eq`ual to each other.
--}
+{- | Check if an 'Interval' is empty. -}
 isEmpty :: (Enum a, Ord a) => Interval a -> Bool
-isEmpty (Interval (LowerBound v1 in1) (UpperBound v2 in2)) = case v1 `compare` v2 of
-    LT -> if openInterval then checkEnds v1 v2 else False
+isEmpty (Interval lb ub) = case strictLowerValue lb `compare` strictUpperValue ub of
+    -- We have at least two possible values, the lower bound and the upper bound
+    LT -> False
+    -- We have one possible value, the lower bound/upper bound
+    EQ -> False
+    -- We have no possible values
     GT -> True
```

Comment:
I think this is equivalent to:

```
isEmpty (Interval lb ub) = strictLowerValue lb > strictUpperValue ub
```

, but it's ok for me to leave it as it is for clarity

Reply:
Yeah, I just had to reason this one out a few times, so I thought I'd be super explicit!